src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,53 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
+    private final Map<Class<?>, Class<?>> implementations =
+            Map.of(
+                    FileReaderService.class, FileReaderServiceImpl.class,
+                    ProductService.class, ProductServiceImpl.class,
+                    ProductParser.class, ProductParserImpl.class
+            );
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Missing Component annotation");
+        }
+        if (instances.containsKey(interfaceClazz)) {
+            return instances.get(interfaceClazz);
+        }
+        try {
```

Comment:
too abstract exception handling. try to split to more concrete ones: check for IllegalAccessException while field.set; check for ReflectiveOperationException while creating new instance